src/compiler/parser.ts

```diff
@@ -197,8 +198,12 @@ export type AST =
 // Parser
 // -----------------------------------------------------------------------------
 const cache: WeakMap<Element, AST> = new WeakMap();
+let customDirectives: customDirectives = {};
 
-export function parse(xml: string | Element): AST {
+export function parse(xml: string | Element, customDir?: customDirectives): AST {
+  if (customDir) {
```

Comment:
this is really ugly. is there a reason not to use the parsing context? you can add information to it